File: app/src/main/java/net/veldor/flibustaloader/ui/BookmarksActivity.java
package net.veldor.flibustaloader.ui;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import net.veldor.flibustaloader.App;
import net.veldor.flibustaloader.R;
import net.veldor.flibustaloader.adapters.BookmarksAdapter;
import net.veldor.flibustaloader.database.entity.Bookmark;

import java.util.List;

public class BookmarksActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.new_bookmarks_activity);
        setupInterface();
    }

    @Override
    protected void setupInterface() {
        super.setupInterface();
        // скрою переход на данное активити
        Menu menuNav = mNavigationView.getMenu();
        MenuItem item = menuNav.findItem(R.id.goToBookmarks);
        item.setEnabled(false);
        item.setChecked(true);

        // получу список закладок
        List<Bookmark> bookmarksList = App.getInstance().mDatabase.bookmarksDao().getAllBookmarks();
        RecyclerView recycler = findViewById(R.id.resultsList);
        BookmarksAdapter adapter = new BookmarksAdapter(bookmarksList);
        recycler.setAdapter(adapter);
        recycler.setLayoutManager(new LinearLayoutManager(this));
    }
}

File: app/src/main/java/net/veldor/flibustaloader/ui/BookmarksActivity.kt
package net.veldor.flibustaloader.ui

import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import net.veldor.flibustaloader.App
import net.veldor.flibustaloader.R
import net.veldor.flibustaloader.adapters.BookmarksAdapter
import net.veldor.flibustaloader.databinding.ActivityBookmarksBinding
import net.veldor.flibustaloader.utils.PreferencesHandler

class BookmarksActivity : BaseActivity() {
    private lateinit var binding: ActivityBookmarksBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityBookmarksBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setupInterface()
    }

    override fun setupInterface() {
        super.setupInterface()
        if(PreferencesHandler.instance.isEInk){
            paintToolbar(binding.toolbar)
        }
        binding.toolbar.title = getString(R.string.bookmarks_title)
        // скрою переход на данное активити
        val menuNav = mNavigationView.menu
        val item = menuNav.findItem(R.id.goToBookmarks)
        item.isEnabled = false
        item.isChecked = true

        // получу список закладок
        val bookmarksList = App.instance.mDatabase.bookmarksDao().allBookmarks
        binding.resultsList.layoutManager = LinearLayoutManager(this)
        binding.resultsList.adapter = BookmarksAdapter(bookmarksList)
    }
}